mobile/ios/ios-emulator-services.ts

```diff
@@ -202,15 +202,16 @@ class IosEmulatorServices implements Mobile.IiOSSimulatorService {
 			
 			let runningSimulatorId = this.getRunningSimulatorId(appIdentifier).wait();
 			let applicationPath = this.getApplicationPath(appIdentifier, runningSimulatorId).wait();
+			let applicationName = _.last(applicationPath.split(path.sep));
 			syncAction(applicationPath);
 		
 			try {
-				this.$childProcess.exec("killall launchd_sim").wait();
-				this.$childProcess.exec(`xcrun simctl launch ${runningSimulatorId} ${appIdentifier}`).wait();				
+				this.$childProcess.exec(`killall ${applicationName.split(".")[0]}`).wait();					
 			} catch(e) {
 				this.$logger.trace("Unable to kill simulator: " + e);
```

Comment:
If killall fails, we'll go in the strange and inconsistent state when the button will have multiple contents from different running processes. Shouldn't we just fail directly here?